src/serverExplorer.ts

```diff
@@ -37,6 +37,29 @@ export class ServersViewTreeDataProvider implements TreeDataProvider<Protocol.Se
         this.refresh();
     }
 
+    restartServer(value) {
+        const status: number = this.serverStatus.get(value.id);
+        const serverstatus =  this.serverStatusEnum.get(status);
+        const params: Protocol.LaunchParameters = {
+            mode: 'run',
+            params: {
+                id: value.id,
+                    serverType: value.type,
+                    attributes: new Map<string, any>()
+                }
+            };
+            if (serverstatus === 'Stopped') {
+                this.client.startServerAsync(params);
+            } else {
+                return this.client.stopServerAsync({ id: value.id, force: false })
+                .then(()=>{
+                    setTimeout(() => {
```

Comment:
timeout is not required here, just call `this.client.startServerAsync(params)` and return resulting promise.